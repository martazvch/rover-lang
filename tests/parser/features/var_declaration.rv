var home: House
var home = 12 + 4.
var home: House = 12 + 4.
