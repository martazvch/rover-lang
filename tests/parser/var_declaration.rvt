// Tests variable declaration

code:
var home

expect:
[Var declaration home, type void, value
    none
]

================

code:
var home: House

expect:
[Var declaration home, type House, value
    none
]

================

code:
var home = 12 + 4.

expect:
[Var declaration home, type void, value
    [Binop +]
        [Int literal 12]
        [Float literal 4.]
]

================

code:
var home: House = 12 + 4.

expect:
[Var declaration home, type House, value
    [Binop +]
        [Int literal 12]
        [Float literal 4.]
]

================

code:
var
var : House
var home House
var home : 12+4
var home : House = var a

error:
ExpectName, variable
ExpectName, variable
ExpectColonBeforeType
ExpectTypeName
ExpectExpr, var

================
