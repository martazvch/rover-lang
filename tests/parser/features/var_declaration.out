{
    "var_decl": {
        "name": "home",
        "type": {},
        "value": {}
    },
    "var_decl": {
        "name": "home",
        "type": "House",
        "value": {}
    },
    "var_decl": {
        "name": "home",
        "type": {},
        "value": {
            "binop": {
                "lhs": {
                    "literal": "12"
                },
                "rhs": {
                    "literal": "4."
                },
                "op": "+"
            }
        }
    },
    "var_decl": {
        "name": "home",
        "type": "House",
        "value": {
            "binop": {
                "lhs": {
                    "literal": "12"
                },
                "rhs": {
                    "literal": "4."
                },
                "op": "+"
            }
        }
    }
}
